### PR TITLE
[Developer] Add a Welcome page to the workflow for Keyman Developer.

### LINKED ISSUE #2123: [Developer] When Keyman Developer starts, show a welcome page that encourages creating a project
**Is your feature request related to a problem? Please describe.**
It's not clear what a user should do when they start Keyman Developer for the first time. Nowadays, they should create a project!

**Describe the solution you'd like**
When Keyman Developer starts, show a welcome page that encourages creating a project... and discourages creating files separately.

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #2123.

This changes the startup flow for Keyman Developer to guide the user into creating a project as a starting point
rather than the current situation where the project feels kinda optional. This does result in a number of changes:

1. The addition of a state where no project is active (IsGlobalProjectUIReady = nil).
2. Addition of a Welcome page that shows version info, recent projects, and guides the user into creating a project.
3. Removal of the now-redundant Splash screen.
4. The addition of a Close Project menu item.

The new Welcome screen looks like this:

image

Updated Options dialog:

image

### CODE REVIEW

File: windows/src/developer/TIKE/actions/dmActionsMain.pas [hunk 1]
```diff
@@ -935,19 +984,48 @@ procedure TmodActionsMain.actWindowPrevUpdate(Sender: TObject);
 procedure TmodActionsMain.DataModuleCreate(Sender: TObject);
 begin
   actFileOpen.Dialog.Filter :=
```
Comment (mcdurdin) [inline windows/src/developer/TIKE/actions/dmActionsMain.pas]: Fixed up the filter for File Open when I saw it...

File: windows/src/developer/TIKE/actions/dmActionsMain.pas [hunk 2]
```diff
@@ -560,13 +592,27 @@ procedure TmodActionsMain.OpenProject(FileName: WideString);
...
   frmKeymanDeveloper.UpdateCaption;
 end;
 
+procedure TmodActionsMain.CloseProject;
+begin
+  if IsGlobalProjectUIReady then
+  begin
+    if not SaveAndCloseAllFiles then Exit;
+  end;
+  FreeGlobalProjectUI;
```
Comment (darcywong00) [inline windows/src/developer/TIKE/actions/dmActionsMain.pas]: Is it safer to move `FreeGlobalProjectUI` inside the `if` block as done in ll. 585 and 598?

**Comment by mcdurdin** [comment]
@DavidLRowe, @dhigby, I think this workflow change that we implemented today should help with the project-oriented workflow for Keyman Developer that we essentially require these days. Would appreciate any feedback you can give (feel free to tag other people if you think of them).

File: windows/src/developer/TIKE/actions/dmActionsMain.pas (hunk 1 above)
Comment (darcywong00) [reply windows/src/developer/TIKE/actions/dmActionsMain.pas]: Nice! This fixes the first point of #2127

**Comment by DavidLRowe** [comment]
@mcdurdin Changes look good (though will mean some rewriting of tutorials). Can you tell me in what beta version of Keyman Developer this change will appear?

File: windows/src/developer/TIKE/actions/dmActionsMain.pas (hunk 2 above)
Comment (mcdurdin) [reply windows/src/developer/TIKE/actions/dmActionsMain.pas]: It won't matter because it will be a no-op but will move for consistency :)

**Comment by mcdurdin** [comment]
@DavidLRowe this will be released in build 12.0.39.

**Pull request merged**